### DISCUSSION TITLE
How to set polymorphic fields in fields.yaml

### DISCUSSION THREAD

**Discussion by SledgehammerPL** [wintercms/winter#760] [Q&A]
from tutorial:

```
class Comment extends Model
{
    public $morphTo = [
        'commentable' => []
    ];
}

class Post extends Model
{
    public $morphMany = [
        'comments' => [\Acme\Blog\Models\Comment::class, 'name' => 'commentable']
    ];
}

class Product extends Model
{
    public $morphMany = [
        'comments' => [\Acme\Blog\Models\Comment::class, 'name' => 'commentable']
    ];
}
```

And now - can I go to Comment Model and choose Post or Product, then choose some Post and comment it? How to code it in fields.yaml of Model Comment?

   ```
 commentable:
        type: relation
```

causes:
`The "morphTo" relation type is unsupported by the Relation widget.`

**Comment by bennothommo** [comment]
I think the error is pretty clear there - the `morphTo` relation isn't supported by the Relation widget yet. You will need to use another relation type.

**Reply by SledgehammerPL** [reply]
Thank you, Capt. Obvious... Maybe little help with choosing the another relation type?

**Reply by bennothommo** [reply]
I wasn't trying to be obtuse, I was simply stating that you need to use a different relation type. I don't know your use case, beyond that you seem to want to allow comments between posts and products.

You could just create a `PostComment` model and use a `hasMany` relation in `Posts` to point to it. You won't be able to share it with the `Product` model, but that's the trade-off if you want to use it with the Relation widget.

**Reply by SledgehammerPL** [reply]
No no, no hasMany - this is Polymorphic - morphMany.    hasMany would be easy - but i need polymorphic in this case

**Reply by bennothommo** [reply]
Feel free to submit a PR introducing morphMany support for the Relation widget then.
